android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -30,26 +29,23 @@ open class MainActivity : ReactActivity() {
     /* Returns true just if we did handle the intent. */
     private fun maybeHandleIntent(intent: Intent?): Boolean {
         intent ?: return false
-        val url = intent.data
+        val maybeReactContext = reactApplication.tryGetReactContext()
         when (intent.action) {
             // Share-to-Zulip
             Intent.ACTION_SEND, Intent.ACTION_SEND_MULTIPLE -> {
-                handleSend(intent, reactApplication.tryGetReactContext(), contentResolver)
+                handleSend(intent, maybeReactContext, contentResolver)
                 return true
             }
 
-            Intent.ACTION_VIEW -> when {
-                // Launch MainActivity on tapping a notification
-                url?.scheme == "zulip" && url.authority == NOTIFICATION_URL_AUTHORITY -> {
-                    val data = intent.getBundleExtra(EXTRA_NOTIFICATION_DATA) ?: return false
-                    logNotificationData("notif opened", data)
-                    notifyReact(reactApplication.tryGetReactContext(), data)
+            Intent.ACTION_VIEW -> {
+                if (maybeHandleViewIntent(intent, maybeReactContext)) {
```

Comment:
Does the function `maybeHandleViewIntent` want a name that mentions notifications, so it's clear what it means in the context of surrounding code, which isn't all about notifications?

Reply:
Yeah. How about `maybeHandleViewNotif`? The particular "view" intents that it does handle are those that are about a notification — those where the URL identifying what the user wanted to view is a URL in a form we made up to mean "open a notification".